examples/connect-latency.bpf.c

```diff
@@ -0,0 +1,133 @@
+#include <vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_tracing.h>
+#include <bpf/bpf_core_read.h>
+#include "bits.bpf.h"
+#include "maps.bpf.h"
+
+#define MAX_LATENCY_SLOT 26
+
+struct connect_start_key_t {
+    u64 pid_tgid;
+};
+
+struct connect_start_val_t {
+    u64 ts;
+    int addrlen;
+    u32 d_ip;    // Destination IPv4 address
+    u16 d_port;  // Destination port number
+};
+
+struct connect_latency_key_t {
+    u32 d_ip;    // Destination IPv4 address
+    u16 d_port;  // Destination port number
+    u64 slot;
+};
+
+struct {
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __uint(max_entries, 10240);
+    __type(key, struct connect_start_key_t);
+    __type(value, struct connect_start_val_t);
+} connect_start SEC(".maps");
+
+struct {
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __uint(max_entries, MAX_LATENCY_SLOT + 2);
+    __type(key, struct connect_latency_key_t);
+    __type(value, u64);
+} connect_latency_seconds SEC(".maps");
+
+static inline __u16 ntohs(__u16 value) {
+    return ((value & 0x00FF) << 8) | ((value & 0xFF00) >> 8);
+}
+
+SEC("kprobe/__sys_connect")
+int BPF_KPROBE(kprobe__sys_connect, int sockfd, const struct sockaddr *addr, int addrlen)
+{
+    struct sockaddr sa;
+    struct connect_start_val_t start_val = {};
+
+    bpf_probe_read(&sa, sizeof(sa), addr);
+
+    if (sa.sa_family == 1) {
```

Comment:
There are lots of families and you probably only care about `AF_INET`:

* https://elixir.bootlin.com/linux/v6.6-rc1/source/include/linux/socket.h#L192

I suggest you add `#define AF_INET 2` and use it here.